### PR TITLE
`carthage build` command

### PR DESCRIPTION
**Pull request by uny**
For the issue https://github.com/fastlane/fastlane/issues/1375.

Also I added some options for carthage such as https://github.com/fastlane/fastlane/issues/813.
1. add `command` option to support `carthage build`
2. add options `configuration`, `no_skip_current` and `no_checkout`.
3. change `use_binaries` into `no_use_binaries` to follow the `carthage` itself options.
4. sort `cmd <<`, `FastlaneCore::ConfigItem.new`, and rspec as the same order.

### CODE REVIEW

**Comment by JaviSoto** [comment]
You can do what you wanted to do by running:

`carthage update:false`

This change is not strictly necessary (although the API may be nicer than before), and is a backwards incompatible change.

**Comment by JaviSoto** [comment]
For context, this is the PR I sent recently: https://github.com/fastlane/fastlane/pull/1354

**Comment by uny** [comment]
https://github.com/fastlane/fastlane/pull/1354 is for `carthage bootstrap` and `carthage update`.
We need `carthage build` to use the sources committed in `Carthage/Checkouts`.

**Comment by JaviSoto** [comment]
Oh I see now! Yeah, then I think we need your change.

File: lib/fastlane/actions/carthage.rb [hunk 1]
```diff
@@ -26,39 +29,50 @@ def self.description
...
         [
-          FastlaneCore::ConfigItem.new(key: :update,
-                                       env_name: "FL_CARTHAGE_UPDATE",
-                                       description: "Update the the project's dependencies",
-                                       is_string: false,
+          FastlaneCore::ConfigItem.new(key: :command,
+                                       env_name: "FL_CARTHAGE_COMMAND",
+                                       description: "Carthage command (one of `build`, `bootstrap`, `update`)",
                                        optional: true,
-                                       default_value: false,
```
Comment (JaviSoto) [inline lib/fastlane/actions/carthage.rb]: Instead of having an if/else on the implementation, you could make "bootstrap" the default value here :)

File: lib/fastlane/actions/carthage.rb [hunk 2]
```diff
@@ -68,20 +82,29 @@ def self.available_options
...
                                        verify_block: proc do |value|
-                                         raise "Please pass a valid platform. Use one of the following: all, iOS, Mac, watchOS" unless ["all", "iOS", "Mac", "watchOS"].include? value
+                                         raise "Please pass a valid value for use_submodules. Use one of the following: true, false" unless value.kind_of?(TrueClass) || value.kind_of?(FalseClass)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :no_use_binaries,
+                                       env_name: "FL_CARTHAGE_NO_USE_BINARIES",
+                                       description: "Check out dependency repositories even when prebuilt frameworks exist",
+                                       is_string: false,
+                                       optional: true,
+                                       verify_block: proc do |value|
```
Comment (JaviSoto) [inline lib/fastlane/actions/carthage.rb]: There seem to be a ton more changes in the configuration down here. Mind describing what they are? I'm having trouble following it.

Comment (uny) [inline lib/fastlane/actions/carthage.rb]: I did in this PR:
1. add `command` option to support `carthage build`
2. add options `configuration`, `no_skip_current` and `no_checkout`.
3. `use_binaries` to `no_use_binaries` to follow the `carthage` itself options.
4. Sorting `cmd <<`, `FastlaneCore::ConfigItem.new`, and rspec as the same order.

The most part is 4.

File: lib/fastlane/actions/carthage.rb (hunk 1 above)
Comment (uny) [inline lib/fastlane/actions/carthage.rb]: Sure! I will fix it.

File: lib/fastlane/actions/carthage.rb (hunk 2 above)
Comment (JaviSoto) [inline lib/fastlane/actions/carthage.rb]: What do you think about splitting the changes into separate PRs so that they can be discussed separately?
Personally I'm concerned about the breaking changes. I think it's great to support all the Carthage commands, but the rest of changes seem less important at this point, and it would be beat do discuss in a separate PR to go over a possible depreciation path.

Comment (uny) [inline lib/fastlane/actions/carthage.rb]: > What do you think about splitting the changes into separate PRs so that they can be discussed separately?

Yes, I definitely agree with you.
Just my laziness.

I will make a PR only for `command` option.
I would like to close this PR and make a new one because I want to remember what I did.
Is it OK?

**Comment by uny** [comment]
The new PR is https://github.com/fastlane/fastlane/pull/1394.
